src/apm.js

```diff
@@ -0,0 +1,98 @@
+import { window } from './utils'
+import Config from './config'
+
+function isFloat(n) {
+    return Number(n) === n && n % 1 !== 0
+}
+
+export function optimisePerformanceData(performanceEntries) {
+    performanceEntries.forEach((performanceEntry, index) => {
+        for (const performanceEntryItemKey in performanceEntry) {
+            if (
+                isFloat(performanceEntry[performanceEntryItemKey]) &&
+                performanceEntry[performanceEntryItemKey].toString().match(/^\d+\.\d{4,}$/)
+            ) {
+                performanceEntries[index][performanceEntryItemKey] = Number(
+                    performanceEntry[performanceEntryItemKey].toFixed(3)
+                )
+            }
+
+            if (
+                ['serverTiming', 'workerTiming'].includes(performanceEntryItemKey) &&
+                performanceEntry[performanceEntryItemKey].length === 0
+            ) {
+                delete performanceEntries[index][performanceEntryItemKey]
+            }
+
+            if (performanceEntryItemKey === 'entryType' && performanceEntry[performanceEntryItemKey] === 'resource') {
+                delete performanceEntries[index][performanceEntryItemKey]
+            }
+
+            if (performanceEntryItemKey === 'nextHopProtocol') {
+                delete performanceEntries[index][performanceEntryItemKey]
+            }
+
+            if (performanceEntry[performanceEntryItemKey] === 0) {
+                delete performanceEntries[index][performanceEntryItemKey]
+            }
+        }
+    })
+
+    return deduplicateKeys(performanceEntries)
+}
```

Comment:
Thinking of this more, could there be a case where we should instead of deleting keys, whitelist and pick the keys we want? As in, what's the chance the ~~Web Standards~~ Chrome folk decide to add another 200kb key to the object?

Reply:
Yep, I think once we have the UI stable (assuming we carry through with it) then it makes sense to only send data we're going to use. I'd worry that an allow list now would change frequently...